File: app/models/StatusSocket.js



function StatusSocket(url = 'ws://echo.websocket.org'){
    let socket = new WebSocket(url);
    socket.onopen = () => console.info('Socket Open: ' + url); 
    socket.onclose = () => console.error('Socket Closed: ' + url); 
    socket.onmessage = (message) => console.log( message.data );
    
    return {
        send : (message) => { socket.send(message) },
        receive : (callback) => { 
            socket.onmessage = function received(message){
                callback( JSON.parse(message.data) );
            } 
        },
    };
}

File: server/status-rt-server/src/main/java/br/status/rt/StatusSocket.java
package br.status.rt;

import java.io.IOException;

import javax.websocket.ClientEndpoint;
import javax.websocket.CloseReason;
import javax.websocket.OnClose;
import javax.websocket.OnError;
import javax.websocket.OnMessage;
import javax.websocket.OnOpen;
import javax.websocket.Session;
import javax.websocket.server.ServerEndpoint;

@ClientEndpoint
@ServerEndpoint("/status/")
public class StatusSocket {

	@OnOpen
	public void onOpen(Session session) {
		System.out.println("WebSocket open");
	}

	@OnMessage
	public void onMessage(String message, Session session) {
		System.out.println("Received message: " + message);
		sendBack(message, session);
	}

	@OnClose
	public void onClose(CloseReason reason) {
		System.out.println("WebSocket closed");
	}

	@OnError
	public void onWebSocketError(Throwable cause) {
		cause.printStackTrace(System.err);
	}

	private void sendBack(String message, Session session) {
		try {
			session.getBasicRemote().sendText(message);
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
}
